lisa/runner.py

```diff
@@ -190,6 +193,21 @@ def _reset_awaitable_timer(self, name: str) -> None:
         self._wait_resource_logged = False
         self._wait_resource_timers[name] = _wait_resource_timer
 
+    def _check_retry_limit(self) -> bool:
+        if self._max_retries > 0:
```

Comment:
The retry counter should be on the environment, not a global one. It's different with timeout.

Reply:
do you mean add current_attempt into environment?

Reply:
Yes, it can be called like `tried_times`.

Reply:
updated